### PR TITLE
Add test_category in teeio-validator

### LINKED ISSUE #108: Add test_category in ide_test.ini to support more tests (for example, TDISP, CXL-IDE, etc)

### PR DESCRIPTION
**Pull request by mxu9**
Fix #108 

Add test_category in teeio-validator

### CODE REVIEW

File: teeio-validator/include/helperlib.h [hunk 1]
```diff
@@ -42,6 +42,9 @@ uint8_t calculate_checksum(uint8_t *table, size_t length);
 bool is_valid_topology_type(uint8_t *type);
 IDE_TEST_TOPOLOGY_TYPE get_topology_type_from_name(uint8_t* name);
 
+bool is_valid_ide_test_category(uint8_t *test_category);
```
Comment (jyao1) [inline teeio-validator/include/helperlib.h:45]: why use "ide_test"?

can other test use this one?

File: teeio-validator/include/ide_test.h [hunk 2]
```diff
@@ -66,6 +66,12 @@
 #define CONFIGURATION_SECTION "Configuration_%d"
 #define TEST_SUITE_SECTION "TestSuite_%d"
 
+typedef enum {
+  IDE_TEST_CATEGORY_PCIE_IDE = 0,
+  IDE_TEST_CATEGORY_CXL_IDE,
+  IDE_TEST_CATEGORY_NUM
```
Comment (jyao1) [inline teeio-validator/include/ide_test.h:72]: suggest to use IDE_TEST_CATEGORY_MAX

File: teeio-validator/include/helperlib.h (hunk 1 above)
Comment (mxu9) [reply teeio-validator/include/helperlib.h]: Renamed as is_valid_test_category.

File: teeio-validator/include/ide_test.h (hunk 2 above)
Comment (mxu9) [reply teeio-validator/include/ide_test.h]: Will updated in next version.

**Pull request merged**